src/lib.rs

```diff
@@ -0,0 +1,704 @@
+struct LdkLiteRuntime {
+	_tokio_runtime: tokio::runtime::Runtime,
+	_background_processor: BackgroundProcessor,
+	stop_networking: Arc<AtomicBool>,
+	stop_wallet_sync: Arc<AtomicBool>,
+}
+
+/// The main interface object of the simplified API, wrapping the necessary LDK and BDK functionalities.
+///
+/// Needs to be initialized and instantiated through [`LdkLiteBuilder::build`].
+pub struct LdkLite {
+	running: RwLock<Option<LdkLiteRuntime>>,
+	config: Arc<LdkLiteConfig>,
+	chain_access: Arc<LdkLiteChainAccess<MemoryDatabase>>,
+	channel_manager: Arc<ChannelManager>,
+	chain_monitor: Arc<ChainMonitor>,
+	peer_manager: Arc<PeerManager>,
+	gossip_sync: Arc<GossipSync>,
+	persister: Arc<FilesystemPersister>,
+	logger: Arc<FilesystemLogger>,
+	scorer: Arc<Mutex<Scorer>>,
+	network_graph: Arc<NetworkGraph>,
+	invoice_payer: Arc<InvoicePayer<LdkLiteEventHandler>>,
+	_inbound_payments: Arc<PaymentInfoStorage>,
+	_outbound_payments: Arc<PaymentInfoStorage>,
+	event_queue: (EventSender, EventReceiver),
+}
+
+impl LdkLite {
+	/// Starts the necessary background tasks, such as handling events coming from user input,
+	/// LDK/BDK, and the peer-to-peer network. After this returns, the [`LdkLite`] instance can be
+	/// controlled via the provided API methods in a thread-safe manner.
+	pub fn start(&mut self) -> Result<(), Error> {
+		// Acquire a run lock and hold it until we're setup.
+		// TODO: maybe this could be done cleaner with a Condvar?
+		let mut run_lock = self.running.write().unwrap();
+		if run_lock.is_some() {
+			// We're already running.
+			return Err(Error::AlreadyRunning);
+		}
+
+		let runtime = self.setup_runtime()?;
+		*run_lock = Some(runtime);
+		Ok(())
+	}
+
+	/// Disconnects all peers, stops all running background tasks, and shuts down [`LdkLite`].
+	pub fn stop(&mut self) -> Result<(), Error> {
+		let mut run_lock = self.running.write().unwrap();
+		if run_lock.is_none() {
+			// We're not running.
+			return Err(Error::NotRunning);
+		}
+
+		let runtime = run_lock.as_ref().unwrap();
+
+		// Stop wallet sync
+		runtime.stop_wallet_sync.store(true, Ordering::Release);
+
+		// Stop networking
+		runtime.stop_networking.store(true, Ordering::Release);
+		self.peer_manager.disconnect_all_peers();
+
+		// Wait for 500ms to make sure shutdown can happen cleanly.
+		thread::sleep(Duration::from_millis(500));
+
+		// Drop the runtime, which stops the background processor and any possibly remaining tokio threads.
+		*run_lock = None;
+		Ok(())
+	}
+
+	fn setup_runtime(&self) -> Result<LdkLiteRuntime, Error> {
+		let _tokio_runtime =
+			tokio::runtime::Builder::new_multi_thread().enable_all().build().unwrap();
+
+		// Setup wallet sync
+		let chain_access = Arc::clone(&self.chain_access);
+		let sync_cman = Arc::clone(&self.channel_manager);
+		let sync_cmon = Arc::clone(&self.chain_monitor);
+		let stop_wallet_sync = Arc::new(AtomicBool::new(false));
+		let stop_sync = Arc::clone(&stop_wallet_sync);
+
+		thread::spawn(move || {
```

Comment:
Yeah, fair enough. Thinking was that we want to phase out `tokio` after https://github.com/lightningdevkit/rust-lightning/issues/1493. But your of course right, as long as it is there, we can make use of its lighter weight, even for non-async stuff.

Reply:
Alright, fair enough. I just wasn't too sure on how relying on async stuff might impact compatibility at different points, e.g., as `LdkLite` is a library we can't make *everything* async since we probably don't want to expect the user to always provide a tokio runtime from the getgo. But setting one up and using that, just as we do right now, is probably fine.